src/views/EmergencyReportAndDocument/index.tsx

```diff
@@ -1,17 +1,316 @@
+
+const PAGE_SIZE = 25;
 
 // eslint-disable-next-line import/prefer-default-export
 export function Component() {
     const strings = useTranslation(i18n);
+    const { emergencyResponse } = useOutletContext<EmergencyOutletContext>();
+    const [page, setPage] = useState(1);
+
+    const regions = useRegion();
+
+    const {
+        response: situationReportsResponse,
+    } = useRequest({
+        skip: isNotDefined(emergencyResponse),
+        url: '/api/v2/situation_report/',
+        query: isDefined(emergencyResponse) ? {
+            event: emergencyResponse.id,
+            limit: 1000, // FIXME: we should not use this unbounded request
+        } : undefined, // TODO: we need to add search filter in server
+    });
+
+    const {
+        pending: appealDocumentsPending,
+        response: appealDocumentsResponse,
+    } = useRequest({
+        skip: isNotDefined(emergencyResponse?.appeals)
+           || (isDefined(emergencyResponse)
+            && isDefined(emergencyResponse.appeals)
+            && emergencyResponse.appeals.length < 1),
+        url: '/api/v2/appeal_document/',
+        query: isDefined(emergencyResponse) ? {
+            appeal: emergencyResponse.appeals.map((appeal) => appeal.id).filter(isDefined),
+            limit: PAGE_SIZE,
+            offset: PAGE_SIZE * (page - 1),
+        } : undefined, // TODO: fix typing issue in server
+    });
+
+    const getRegionList = useCallback((regionList: number[]) => (
+        regionList.map((region) => regions?.find((r) => r.id === region))
+            .filter(isDefined)
+    ), [regions]);
+
+    const columns = useMemo(
+        () => ([
+            createDateColumn<FieldReportListItem, number>(
+                'created_at',
+                strings.fieldReportsTableCreatedAt,
+                (item) => item.created_at,
+                {
+                    columnClassName: styles.date,
+                },
+            ),
+            createLinkColumn<FieldReportListItem, number>(
+                'summary',
+                strings.fieldReportsTableName,
+                (item) => item.summary,
+                (item) => ({
+                    to: 'fieldReportDetails',
+                    urlParams: { fieldReportId: item.id },
+                }),
+                {
+                    columnClassName: styles.summary,
+                },
+            ),
+            createCountryListColumn<FieldReportListItem, number>(
+                'countries',
+                strings.fieldReportsTableCountry,
+                (item) => item.countries,
+            ),
+            createRegionListColumn<FieldReportListItem, number>(
+                'regions',
+                strings.fieldReportsTableRegion,
+                (item) => {
+                    const ids = unique(item.countries.map((country) => country.region))
+                        .filter(isDefined);
+                    return getRegionList(ids);
+                },
```

Comment:
This is not very good for performance. Should we instead update the data on the table?

cc: @frozenhelium

Reply:
Yes, and we can also, use `createListDisplayColumn` which has similar usecase